{{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py

```diff
@@ -1,53 +1,75 @@
-import tempfile
-from pathlib import Path
-
 from hamcrest import assert_that, contains, has_properties
-from microcosm_sagemaker.artifact import InputArtifact, OutputArtifact
-from microcosm_sagemaker.input_data import InputData
-from microcosm_sagemaker.testing.directory_comparison import (
-    directory_comparison,
+from microcosm_sagemaker.testing.bundle import (
+    BundleFitTestCase,
+    BundleLoadTestCase,
+    BundleSaveTestCase,
+    BundleTestCase,
 )
 
-from {{ cookiecutter.package_name }}.app_hooks.train.app import create_app
+from {{ cookiecutter.package_name }}.bundles.example import ExampleBundle
 from {{ cookiecutter.package_name }}.tests.fixtures import get_fixture_path
 
 
-class TestExampleBundle:
-    def setup(self) -> None:
-        self.graph = create_app(extra_deps=["example_bundle"])
-        self.training_initializers = self.graph.training_initializers
+class ExampleBundleTestCase(BundleTestCase):
+    """
+    All of the test cases in this file will be testing example bundle, and use
+    the same root input artifact.  They will also check the bundle prediction
+    in the same way when necessary.  Thus, we define these parameters in a base
+    class here.
+
+    """
 
-        self.input_data = InputData(get_fixture_path("example_input_data"))
-        self.input_artifact = InputArtifact(get_fixture_path("example_input_artifact"))
-        self.gold_output_artifact_path = get_fixture_path("example_gold_output_artifact")
+    # Determines which bundle will be tested
+    bundle_name = "example_bundle"
 
-    def check_bundle_prediction(self) -> None:
+    # Contains a full root input artifact that will be used to automatically
+    # load all dependency bundles and, for the load test, to test loading of
+    # the bundle under test
+    root_input_artifact_path = get_fixture_path("example_input_artifact")
+
+    # This function will be called on the bundle after fitting / loading to
+    # make sure it works correctly
+    def check_bundle_prediction(self, bundle: ExampleBundle) -> None:
         assert_that(
-            self.graph.example_bundle.predict(1.0),
+            bundle.predict(1.0),
```

Comment:
`microcosm-sagemaker` is the one that passes the `bundle` argument to this function, so the only things we have control over in here are: a) the input to the `predict` method and b) the expected output. With that in mind, how about we define a set of `(input, expected_output)` pairs in here; and abstract away the `bundle.predict`? 

For example:
https://github.com/globality-corp/ml-nlp-serviceoffering-classifier/blob/master/ml_nlp_serviceoffering_classifier/tests/bundles/test_service_offering_bundle.py#L31
https://github.com/globality-corp/ml-nlp-serviceoffering-classifier/blob/master/ml_nlp_serviceoffering_classifier/tests/bundles/test_classifier_baseline_bundle.py#L23

Reply:
Nice call.  Just took a swing at that; lmk what you think

Reply:
Looks great!